src/alarms_agent.cpp

```diff
@@ -107,6 +111,37 @@ int main (int argc, char **argv)
       case OPT_COMMUNITY:
         community = optarg;
         break;
+      case OPT_NOTIFICATION_TYPE:
+        {
+          std::vector<std::string> notification_types;
+          Utils::split_string(std::string(optarg), ',', notification_types);
+          for (std::vector<std::string>::iterator it = notification_types.begin();
+               it != notification_types.end();
+               ++it)
+          {
+            if (*it == "rfc3877" && std::find(snmp_notifications.begin(), 
+                                              snmp_notifications.end(),
+                                              NotificationType::RFC3877) == snmp_notifications.end())
+            {
+              snmp_notifications.push_back(NotificationType::RFC3877);
+            }
+            else if (*it == "enterprise" && std::find(snmp_notifications.begin(), 
+                                              snmp_notifications.end(),
+                                              NotificationType::ENTERPRISE) == snmp_notifications.end())
+            {
+              snmp_notifications.push_back(NotificationType::ENTERPRISE);
+            }
+            else
+            {
+              puts("Invalid config option used for snmp notification type");
+            }
+          }
+          if (snmp_notifications.empty())
```

Comment:
Actually, this is in the wrong place as it means snmp_notifications is only populated if a user passes in something under snmp_notification_type. I assume you didn't see this in your testing as you've set the default in the wrapper script as well. I think you should take the default out of the wrapper script and move this check until after all the parameters have been parsed.